Cargo.toml

```diff
@@ -159,6 +159,15 @@ homepage = "https://anza.xyz/"
 license = "Apache-2.0"
 edition = "2021"
 
+[workspace.lints.rust.unexpected_cfgs]
+level = "warn"
+check-cfg = [
+    'cfg(target_os, values("solana"))',
+    'cfg(feature, values("frozen-abi", "no-entrypoint"))',
+    'cfg(RUSTC_WITH_SPECIALIZATION)',
+    'cfg(RUSTC_WITHOUT_SPECIALIZATION)',
+]
+
```

Comment:
nit: this can be a follow up pr, but it seems it's prime time for us to finally move lints from `scripts/cargo-clippy-nightly.sh`

for that, i think this formatting is preferred:

```toml
[workspace.lints.rust]
unexpected_cfgs = { level = "warn", check-cfg = [
  ...
] }
```

like this: https://github.com/anza-xyz/agave/pull/2487/files#r1718380391